scipy/sparse/linalg/_isolve/iterative.py

```diff
@@ -3,18 +3,19 @@
 from scipy.sparse.linalg._interface import LinearOperator
 from .utils import make_system
 from scipy.linalg import get_lapack_funcs
+from scipy._lib.deprecation import _NoValue
 
 __all__ = ['bicg', 'bicgstab', 'cg', 'cgs', 'gmres', 'qmr']
 
 
-def _get_atol(name, b, tol=None, atol=0., rtol=1e-5):
+def _get_atol(name, b, tol=_NoValue, atol=0., rtol=1e-5):
     """
     A helper function to handle tolerance deprecations and normalization
     """
-    if tol is not None:
+    if tol is not _NoValue:
         msg = (f"'scipy.sparse.linalg.{name}' keyword argument 'tol' is "
                "deprecated in favor of 'rtol' and will be removed in SciPy "
-               "v.1.14.0. Until then, if set, will override 'rtol'.")
+               "v.1.14.0. Until then, if set, it will override 'rtol'.")
         warnings.warn(msg, category=DeprecationWarning, stacklevel=4)
         rtol = float(tol)
```

Comment:
I think this needs a minor tweak too as I could call `bicg(A, b, None, None,...........)` and this would fail trying to cast None to a float.
```suggestion
        rtol = float(tol) if tol is not None else rtol
```

Reply:
Yes but before that you will get a warning right? I think the context would be obvious if you ask me but I can add it too, I just don't want to promote the float setting to tol if the error clears

Reply:
Actually, this would indeed fail unexpectedly so let's get it in anyways